README.md

```diff
@@ -10,41 +9,30 @@ Copyright © 2004-2008 Timo Hirvonen <tihirvon@gmail.com>
+Some common autoconf-style options like `--prefix` are also available.
 
 After running configure you can see from the generated `config.mk` file
 what features have been configured in (see the `CONFIG_*` options).
 
-*Note*: For some distributions you need to install development versions
-of the dependencies.  For example if you want to use 'mad' input plugin
-(mp3) you need to install `libmad0-dev` (Debian) or `libmad-devel` (RPM)
-package. After installing dependencies you need to run `./configure`
-again, of course.
-
-If you want to use the Tremor library as alternative for decoding
-Ogg/Vorbis files you have to pass `CONFIG_TREMOR=y` to the configure
-script:
+The packages containing dependencies on common distributions are listed below. All dependencies other than pkg-config and ncurses, iconv, and elogind/systemd are for optional input/output plugins. It is assumed that libc headers, a C compiler, git, and GNU Make are available.
 
-    $ ./configure CONFIG_VORBIS=y CONFIG_TREMOR=y
+<table><tbody>
+<tr><th align="left">Debian/Ubuntu</th><td>apt install pkg-config libncursesw5-dev libfaad-dev libao-dev libasound2-dev libcddb2-dev libcdio-cdda-dev libcue-dev libdiscid-dev libavformat-dev libavcodec-dev libswresample-dev libflac-dev libjack-dev libmad0-dev libmodplug-dev libmpcdec-dev libsystemd-dev libopusfile-dev libpulse-dev libsamplerate0-dev libsndio-dev libvorbis-dev libwavpack-dev</td></tr>
+<tr><th align="left">Fedora/RHEL</th><td>dnf install 'pkgconfig(ncursesw)' 'pkgconfig(alsa)' 'pkgconfig(ao)' 'pkgconfig(libcddb)' 'pkgconfig(libcdio_cdda)' 'pkgconfig(libcue)' 'pkgconfig(libdiscid)' 'pkgconfig(libavformat)' 'pkgconfig(libavcodec)' 'pkgconfig(libswresample)' 'pkgconfig(flac)' 'pkgconfig(jack)' 'pkgconfig(mad)' 'pkgconfig(libmodplug)' libmpcdec-devel 'pkgconfig(libsystemd)' 'pkgconfig(opusfile)' 'pkgconfig(libpulse)' 'pkgconfig(samplerate)' 'pkgconfig(vorbisfile)' 'pkgconfig(wavpack)'
+</td></tr>
+<tr><th align="left">+ RPMFusion</th><td>dnf install faad2-devel libmp4v2-devel</td></tr>
+<tr><th align="left">Arch Linux</th><td>pacman -S pkg-config ncurses libiconv faad2 alsa-lib libao libcddb libcdio-paranoia libcue libdiscid ffmpeg flac jack libmad libmodplug libmp4v2 libmpcdec systemd opusfile libpulse libsamplerate libvorbis wavpack</td></tr>
+<tr><th align="left">Alpine</th><td>apk add pkgconf ncurses-dev gnu-libiconv-dev alsa-lib-dev libao-dev libcddb-dev libcue-dev ffmpeg-dev flac-dev jack-dev libmad-dev libmodplug-dev elogind-dev opus-dev opusfile-dev pulseaudio-dev libsamplerate-dev libvorbis-dev wavpack-dev</td></tr>
+<tr><th align="left">Termux</th><td>apt install libandroid-support ncurses libiconv ffmpeg libmad libmodplug opusfile pulseaudio libflac libvorbis libwavpack</td></tr>
+<tr><th align="left">Homebrew</th><td>brew install pkg-config ncurses faad2 libao libcddb libcdio libcue libdiscid ffmpeg flac jack mad libmodplug mp4v2 musepack opusfile libsamplerate libvorbis wavpack</td></tr>
+</tbody></table>
```

Comment:
It formats a little differently but you can / probably should use the table syntax
```
Distro | Dependencies
:---   | :---
x | apt install blahh
```
Comparing against the arch package, you're additionally listing:
- libcddb - but I guess we can
- libcue - at a glance we don't use it?
- libiconv - uses iconv from libc, not sure what the difference is
- pkg-config - could assume it's installed
- systemd - I guess for dbus? but I don't think we should be listing itp

For Fedora, is the pkgconfig syntax needed? Would be nice to make the list more compact.

Reply:
I considered the table syntax, but this looks nice rendered (though it is uglier raw)... which do you like best @flyingmutant?

Regarding libcue, I think you're right... the last usage of libcue was removed in 85237ec35370f0837c8be3cd1503f9164dbf6703, albeit incompletely. I'll remove it and probably open another PR to remove it entirely.

Regarding libiconv, I'll look at it closer and see if it's actually required.

I listed pkg-config on Arch since I did for the others.

Regarding systemd, it is for dbus. I think we should list it given that we've had bug reports where people are wondering why MPRIS isn't working caused by this.

The pkgconfig syntax is the standard way to do build deps on modern RPM-based distros (the actual package providing it can be split without affecting anything else). Runtime dependencies are based on the compiled ELF sonames.

Reply:
> libiconv

You're right about it; it's included in libc. The libiconv package seems to only exist for compatibility with programs which explicitly link or load libiconv. Not including it for cmus is the correct thing to do to prevent an unnecessary dependency from being added to the binary.

> libcue

I'll remove it from the README and open another PR for it.

Reply:
Done.